src/probability_theory/stopping.lean

```diff
@@ -306,4 +309,154 @@ end linear_order
 
 end is_stopping_time
 
+section linear_order
+
+
+/-- Given a map `u : ι → α → E`, its stopped process with respect to the stopping
+time `τ` is the map `(i, x) ↦ u (min i (τ x)) x`.
+Intuitively, the stopped process stop evolving once the stopping time has occured. -/
+def stopped_process [linear_order ι] (u : ι → α → β) (τ : α → ι) : ι → α → β :=
+λ i x, u (linear_order.min i (τ x)) x
```

Comment:
I wonder if this is worth writing with something like `set.piecewise`, to get some API for free.

Reply:
I'm not sure what API would be useful from `set.piecewise`. Do you have anything in mind?

Reply:
I'm not too sure, honestly. I just felt like there's probably tools for dealing with families of functions, although I may be wrong

Reply:
Maybe someone else can chime in but I feel this isn't that big of a deal